hawkbit-dmf/hawkbit-dmf-amqp/src/test/java/org/eclipse/hawkbit/integration/AmqpMessageDispatcherServiceIntegrationTest.java

```diff
@@ -603,4 +615,93 @@ private static boolean isDownloadAndUpdateRequest(final DmfMultiActionElement mu
+    void assertBatchAndMultiAssignmentsNotCompatible() {
+        enableBatchAssignments();
+        assertThatExceptionOfType(TenantConfigurationValueChangeNotAllowedException.class)
+                .isThrownBy(() -> enableMultiAssignments());
+        disableBatchAssignments();
+
+        enableMultiAssignments();
+        assertThatExceptionOfType(TenantConfigurationValueChangeNotAllowedException.class)
+                .isThrownBy(() -> enableBatchAssignments());
+    }
+
+    @ParameterizedTest
+    @EnumSource(names = { "BATCH_DOWNLOAD_AND_INSTALL", "BATCH_DOWNLOAD" })
+    @Description("Verify payload of batch assignments.")
+    @ExpectEvents({ @Expect(type = TargetCreatedEvent.class, count = 3),
+            @Expect(type = TargetAssignDistributionSetEvent.class, count = 1),
+            @Expect(type = ActionCreatedEvent.class, count = 3), @Expect(type = ActionUpdatedEvent.class, count = 0),
+            @Expect(type = SoftwareModuleCreatedEvent.class, count = 3),
+            @Expect(type = SoftwareModuleUpdatedEvent.class, count = 6),
+            @Expect(type = DistributionSetCreatedEvent.class, count = 1),
+            @Expect(type = TargetUpdatedEvent.class, count = 3), @Expect(type = TargetPollEvent.class, count = 3),
+            @Expect(type = TenantConfigurationCreatedEvent.class, count = 1) })
+    void assertBatchAssignmentsMessagePayload(final EventTopic topic) {
+        enableBatchAssignments();
+
+        final List<String> targets = Arrays.asList("batchCtrlID1", "batchCtrlID2", "batchCtrlID3");
+        for (int i = 0; i < targets.size(); i++) {
+            registerAndAssertTargetWithExistingTenant(targets.get(i), i+1);
+        }
+
+        final DistributionSet ds = testdataFactory.createDistributionSet();
+        testdataFactory.addSoftwareModuleMetadata(ds);
+
+        assignDistributionSet(ds.getId(), targets, topic == BATCH_DOWNLOAD?DOWNLOAD_ONLY:FORCED);
+
+        waitUntilEventMessagesAreDispatchedToTarget(topic);
+
+        final Message message = replyToListener.getLatestEventMessage(topic);
+        final Map<String, Object> headers = message.getMessageProperties().getHeaders();
+        assertThat(headers).containsEntry("type", EVENT.toString());
+        assertThat(headers).containsEntry("topic",topic.toString());
+
+        final DmfBatchDownloadAndUpdateRequest batchRequest = (DmfBatchDownloadAndUpdateRequest) getDmfClient()
+                .getMessageConverter().fromMessage(message);
+
+        assertThat(batchRequest).isExactlyInstanceOf(DmfBatchDownloadAndUpdateRequest.class);
+        assertDmfBatchDownloadAndUpdateRequest(batchRequest, ds.getModules(), targets);
+    }
+
+    protected void assertDmfBatchDownloadAndUpdateRequest(final DmfBatchDownloadAndUpdateRequest request,
+                                                     final Set<SoftwareModule> softwareModules,
+                                                     final List<String> controllerIds) {
+        assertThat(softwareModules)
```

Comment:
could you please use the previously added `assertSoftwareModules` method from `AbstractAmqpServiceIntegrationTest`

Reply:
done